cli/formatter/format.go

```diff
@@ -0,0 +1,752 @@
+			markdownTable = insertStrToStrSlice(markdownTable, 1,
+				constructMarkdownHeaderSeparatorRow(rowsAmount))
+		}
+		var res string
+		for _, v := range markdownTable {
+			res = res + v + "\n"
+		}
+		return res + "\n"
+	}
+	if opts.TableDialect == JiraTableDialect {
+		return t.RenderMarkdown() + "\n\n"
+	}
+
+	return t.Render() + "\n"
+}
+
+// castRenderNodeToMap casts renderNode to map[string]renderNode.
+func castRenderNodeToMap(node renderNode) (map[string]renderNode, error) {
+	var castedRes map[string]renderNode
+	var err error
+	nodeYaml, err := yaml.Marshal(node)
+	if err != nil {
+		return nil, err
+	}
+	castedRes, err = decodeYamlMap(string(nodeYaml))
+	if err != nil {
+		return nil, err
+	}
+
+	return castedRes, nil
+}
+
+// renderArrs returns table as string for arrays case.
+func renderArrs(renderBatch []renderNode, opts *Opts) (string, error) {
+	t := table.NewWriter()
+	t.AppendRow(constructHeaderRowForArrs(renderBatch))
+
+	var rows []table.Row
+	for _, node := range renderBatch {
+		nodeBytes, err := yaml.Marshal(node)
+		if err != nil {
+			return "", err
+		}
+		nodeIterable, err := decodeYamlArr(string(nodeBytes))
+		if err != nil {
+			return "", err
+		}
+		var row []interface{}
+		for _, rowColVal := range nodeIterable {
+			if getRenderNodeType(rowColVal) == scalarRenderNodeType {
+				switch rowColVal.(type) {
+				case float64, float32:
+					row = append(row, strconv.FormatFloat(rowColVal.(float64), 'f', -1, 64))
+				case nil:
+					row = append(row, "nil")
+				default:
+					row = append(row, fmt.Sprintf("%v", rowColVal))
+				}
+			} else {
+				// Encoding as json value for map/array table cell cases.
+				jsonVal, err := json.Marshal(convertMapIntrfToMapStr(rowColVal))
+				if err != nil {
+					return "", err
+				}
+				row = append(row, string(jsonVal))
+			}
+		}
+		rows = append(rows, row)
+	}
+	t.AppendRows(rows)
+	t.Style().Options.SeparateRows = true
+	var rowsAmount = t.Length()
+	if opts.TransposeMode {
+		t.TransposeAppendedRows()
+	}
+	if opts.NoGraphics {
+		t.SetStyle(table.Style{Box: table.StyleWithoutGraphics})
+	}
+	if opts.ColWidthMax > 0 {
+		colWidthTransformer := text.Transformer(func(val interface{}) string {
+			str := fmt.Sprintf("%v", val)
+			widthMax := t.GetColumnConfig()[0].WidthMax
+			if utf8.RuneCountInString(str) > widthMax {
+				firstLine := string([]rune(str)[:widthMax])
+				remainingLines := string([]rune(str)[widthMax:])
+				firstLine = firstLine + "+"
+				remainingLines = text.InsertEveryN(remainingLines, '+', widthMax-1)
+				return firstLine + remainingLines
+			}
+			return fmt.Sprintf("%v", val)
+		})
+		var columnConfigs []table.ColumnConfig
+		for i := 1; i <= len(constructHeaderRowForArrs(renderBatch)); i++ {
+			columnConfigs = append(columnConfigs,
+				table.ColumnConfig{
+					Number:      i,
+					Transformer: colWidthTransformer,
+					WidthMax:    opts.ColWidthMax,
+				},
+			)
+		}
+		t.SetColumnConfigs(columnConfigs)
+	}
```

Comment:
Could you please explain the code with `WidthMax`. Does `t.GetColumnConfig()[0].WidthMax` == `opts.ColWidthMax`?
If this is not the case, then what values can be `t.GetColumnConfig()[0].WidthMax`?

Reply:
Simplified, now using just `opts.ColWidthMax`, it is the same